core/src/main/java/org/infinispan/notifications/cachelistener/filter/KeyValueFilterConverterAsCacheEventFilterConverter.java

```diff
@@ -0,0 +1,60 @@
+package org.infinispan.notifications.cachelistener.filter;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import org.infinispan.commons.marshall.Externalizer;
+import org.infinispan.commons.marshall.SerializeWith;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.filter.KeyValueFilterConverter;
+import org.infinispan.metadata.Metadata;
+
+/**
+ * {@link CacheEventFilterConverter} that uses an underlying {@link KeyValueFilterConverter} to do the conversion and
+ * filtering. The new value and metadata are used as arguments to the underlying filter converter as it doesn't take
+ * both new and old.
+ * @author wburns
+ * @since 9.4
+ */
+@SerializeWith(value = KeyValueFilterConverterAsCacheEventFilterConverter.KVFCACEFCExternalizer.class)
+public class KeyValueFilterConverterAsCacheEventFilterConverter<K, V, C> implements CacheEventFilterConverter<K, V, C> {
+   private final KeyValueFilterConverter<K, V, C> keyValueFilterConverter;
+
+   public KeyValueFilterConverterAsCacheEventFilterConverter(KeyValueFilterConverter<K, V, C> keyValueFilterConverter) {
+      this.keyValueFilterConverter = keyValueFilterConverter;
+   }
+
+   @Override
+   public C filterAndConvert(K key, V oldValue, Metadata oldMetadata, V newValue, Metadata newMetadata, EventType eventType) {
+      return keyValueFilterConverter.convert(key, newValue, newMetadata);
+   }
+
+   @Override
+   public C convert(K key, V oldValue, Metadata oldMetadata, V newValue, Metadata newMetadata, EventType eventType) {
+      return keyValueFilterConverter.convert(key, newValue, newMetadata);
+   }
+
+   @Override
+   public boolean accept(K key, V oldValue, Metadata oldMetadata, V newValue, Metadata newMetadata, EventType eventType) {
+      return keyValueFilterConverter.accept(key, newValue, newMetadata);
+   }
+
+   @Inject
+   protected void injectDependencies(ComponentRegistry cr) {
+      cr.wireDependencies(keyValueFilterConverter);
+   }
+
+   public static class KVFCACEFCExternalizer implements Externalizer<KeyValueFilterConverterAsCacheEventFilterConverter> {
```

Comment:
Could be named `Externalizer`, and it should extend `AdvancedExternalizer` to avoid writing the class name to the stream.

Reply:
Sure I was adding one, but then thought about it more that this is a very minor used operation (scales only with number of listeners - which should be small). But I can add in an ID for it since it seems we want to :)

Reply:
And I can name it `Externalizer` only when it extends `AdvancedExternalizer` (unless I added the entire package name). Just FYI.